### PR TITLE
fix(AV-1695): apiset vocabulary for apisets

### PR DESCRIPTION
**Pull request by bertvi**
fix(AV-1695): apiset vocabulary for apisets

### CODE REVIEW

**Comment by Zharktas** [comment]
You need to use [semantic release](https://github.com/semantic-release/semantic-release) commit messages, otherwise containers will not build. eg. "fix(ticket-number): some message" for bugs and "feat(ticket-number): some other message" for new features.

File: modules/ckanext-ytp_main/ckanext/ytp/plugin.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def dataset_facets(self, facets_dict, package_type):
         lang = get_lang_prefix()
         facets_dict = OrderedDict()
         facets_dict.update({'vocab_international_benchmarks': _('International Benchmarks')})
-        facets_dict.update({'collection_type': _('Collection Types')})
+        facets_dict.update({'collection_type': _('Apiset') if package_type == 'apiset' else _('Collection Types')})
```
Comment (Zharktas) [inline modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: Why is this changed ? The design has different facet filters anyway, so this change does not seem to make sense.

File: modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html [hunk 2]
```diff
@@ -13,12 +13,18 @@
 
 #}
 
-{% if type == 'dataset' or type == 'showcase_dataset' or type == 'harvest' %}
+{% if (type == 'dataset' or type == 'showcase_dataset' or type == 'harvest') and request.path == '/dataset' %}
```
Comment (Zharktas) [inline modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: This check will probably break things.

File: modules/ckanext-ytp_main/ckanext/ytp/plugin.py (hunk 1 above)
Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: It seems to me, the views in "/dataset" and "/apiset" use the same "Collection Types" translation, which translates to "Dataset", which is wrong in "/apiset" view.

Comment (Zharktas) [reply modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: `collection_type` does not exist in the schema of apis, so it would be need to be removed all together. We will update the filters separately as they will be different from datasets, something like harvest does https://github.com/ckan/ckanext-harvest/blob/8874175bc9c09c40ad822df2101ce14fb957f58f/ckanext/harvest/plugin/__init__.py#L340-L347

File: modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html (hunk 2 above)
Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: Seems to work for me. I have done it this way, in order to separate the translation string of apisets and datasets and this could be done based on the request.path (whether the url has apiset in it or dataset)

File: modules/ckanext-ytp_main/ckanext/ytp/plugin.py (hunk 1 above)
Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: Do you mean I should keep that line as it was? Removing it will not display the translation for datasets either I suppose.

File: modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html (hunk 2 above)
Comment (Zharktas) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: Well the url for showcases would be /showcase and harvests its /harvest, that logic would break those at least.

Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: Fixed and tested locally, works for "harvest", "apiset", "dataset" and "showcase"

Comment (Zharktas) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: Instead of hacky solution of looking at request path which might fail at some point if the same template is used from some other template, we should fix why the type is dataset and not apiset

Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/snippets/search_result_text.html]: Fixed, so that the type for apiset is apiset not dataset

File: modules/ckanext-ytp_main/ckanext/ytp/templates/package/search.html [hunk 3]
```diff
@@ -62,7 +62,7 @@ <h1 class="mb-0">{{ _("Datasets") if dataset_type!='apiset' else _("Apisets") }}
             (_('Source'), 'source asc'),
             (_('Popular'), 'views_recent desc') if g.tracking_enabled else (false, false) ]
           %}
-          {% snippet 'snippets/search_form.html', type='dataset', query=c.q, sorting=sorting, sorting_selected=c.sort_by_selected, count=c.page.item_count, facets=facets, show_empty=request.params, error=c.query_error, query_params=c.fields_grouped, show_advanced_search=true %}
+          {% snippet 'snippets/search_form.html', type='dataset' if dataset_type!='apiset' else 'apiset', query=c.q, sorting=sorting, sorting_selected=c.sort_by_selected, count=c.page.item_count, facets=facets, show_empty=request.params, error=c.query_error, query_params=c.fields_grouped, show_advanced_search=true %}
```
Comment (Zharktas) [inline modules/ckanext-ytp_main/ckanext/ytp/templates/package/search.html]: Or just put `type=dataset_type`

Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/templates/package/search.html]: Good idea, fixed.

File: modules/ckanext-ytp_main/ckanext/ytp/plugin.py (hunk 1 above)
Comment (Zharktas) [reply modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: This change is not needed, yes the UI says Tietoaineisto but this filters will be reimplemented anyhow.

Comment (bertvi) [reply modules/ckanext-ytp_main/ckanext/ytp/plugin.py]: Removed the change

**Review by Zharktas — APPROVED** [review]
